source/slang/slang-ir-redundancy-removal.cpp

```diff
@@ -196,6 +196,9 @@ static bool eliminateRedundantTemporaryCopyInFunc(IRFunc* func)
                     continue;
                 }
 
+                if (destPtr->findDecorationImpl(kIROp_DisableCopyEliminationDecoration))
+                    continue;
+
```

Comment:
I'd like to understand why this is a necessary carve-out to have. I see logic below this point that is trying to catch the cases that would be obviously unsafe due to mutability.